src/h_vialib/secure/expiry.py

```diff
@@ -0,0 +1,81 @@
+"""Functions for calculating expiry."""
+
+from datetime import datetime, timedelta, timezone
+
+# An arbitrary time in the past to quantize our expiry times to
+YEAR_ZERO = datetime(year=2020, month=1, day=1, tzinfo=timezone.utc)
+
+
+def quantized_expiry(max_age, divisions=2):
+    """Create a quantized expiry time.
+
+    The time returned is guaranteed be:
+        * A multiple of `max_age / divisions`
+        * At most `max_age` seconds in the future
+        * At least `max_age * (divisions - 1) / divisions` seconds in the future
+
+    :param max_age: The maximum age to issue (int, or timedelta)
+    :param divisions: How many subdivisions of the max age for quantization
+    :return: A datetime object
+    """
+    max_age = _to_int(max_age)
+    quantization = max_age // divisions
+
+    # Timezone aware dates are annoying, but allow you to include the TZ in a
+    # serialisation, which is required for cookie expiry times
+    delta = datetime.now(tz=timezone.utc) - YEAR_ZERO
+
+    quantized_expire = (delta.total_seconds() // quantization) * quantization
+
+    return YEAR_ZERO + timedelta(seconds=quantized_expire + max_age)
+
+
+def as_expires(expires=None, max_age=None):
```

Comment:
I split this off so anything which has this pattern (like most of our secure things) can have all this fancy conversion and parsing without having to code for it everywhere